Automattic/wp-calypso
# Activity Logs: Activate/Deactivate actions and http handlers

Adds rewind activate/deactivate actions and http middleware.

Part of splitting PR #10862 into merge-size chunks. Follows on from rewind status actions in #14660.

This PR sticks closely to the original #10862, keeping the `_SUCCESS` and `_FAILURE` actions so that we can show UI feedback on the toggle button whilst the requests are in-flight.


**To Test**
* PR contains unit tests
* Real network requests can be fired using following console commands:
```
dispatch( { type: 'REWIND_ACTIVATE_REQUEST', siteId: '{site}' } )
dispatch( { type: 'REWIND_DEACTIVATE_REQUEST', siteId: '{site}' } )
```
*Expected Results*
* Use a jetpack site with vaultpress for successful responses. Note that the `use_rewind` field in the status response should toggle on and off after activate/deactivate. To get the status, use:
```dispatch( { type: 'REWIND_STATUS_REQUEST', siteId: '{site}' } )```
* When using a non-vaultpress site, an error notice should display on activate/deactivate
<img width="564" alt="screen shot 2017-06-06 at 11 53 25" src="https://cloud.githubusercontent.com/assets/7767559/26825926/cd8b70f4-4aae-11e7-9bd4-64a61eef1dd3.png">
